src/astro/files/locations/s3.py

```diff
@@ -0,0 +1,44 @@
+import os
+from typing import Dict, List, Tuple
+from urllib.parse import urlparse, urlunparse
+
+from astro.constants import FileLocation
+from astro.files.locations.base import Location
+from astro.utils.dependencies import BotoSession, s3
+
+
+class S3(Location):
```

Comment:
We might want to make this consistent across Location, Types and Database.

For example, we call it `SqliteDatabase` in https://github.com/astro-projects/astro/pull/320/files

Reply:
i.e. should we call it `S3Location`, `S3` , `S3FileLoc` 🤷 -- no strong opinion but would like it to be consistent across all the three entities

Reply:
I think we S3Location gives it the right context it needs, otherwise, this needs to be inferred from the import path. Will make this change

Reply:
Done.